File: src/test/java/org/micoli/php/builders/TaskBuilder.java
package org.micoli.php.builders;

import org.micoli.php.tasks.configuration.Task;

public class TaskBuilder {
    private final Task task;

    private TaskBuilder() {
        this.task = new Task();
    }

    public static TaskBuilder create() {
        return new TaskBuilder();
    }

    public TaskBuilder withTaskId(String taskId) {
        task.taskId = taskId;
        return this;
    }

    public TaskBuilder withLabel(String label) {
        task.label = label;
        return this;
    }

    public Task build() {
        return task;
    }
}

File: src/test/kotlin/org/micoli/php/builders/TaskBuilder.kt
package org.micoli.php.builders

import org.micoli.php.tasks.configuration.Task

class TaskBuilder private constructor() {
    private val task: Task = Task()

    fun withTaskId(taskId: String): TaskBuilder {
        task.taskId = taskId
        return this
    }

    fun withLabel(label: String): TaskBuilder {
        task.label = label
        return this
    }

    fun build(): Task = task

    companion object {
        @JvmStatic fun create(): TaskBuilder = TaskBuilder()
    }
}
